lib/deb_utils.rb

```diff
@@ -0,0 +1,43 @@
+def extract_deb(file, target = nil)
+  # extract_deb: unarchive .deb files
+  # Usage: extract_deb(<file>, <target (optional)>)
+  #   file: a .deb archive file
+  #   target: (optional) a specified file to extract from archive, should be passed in regex
+  #
+  # Example:
+  #   extract_deb('example.deb', /data\..*/) # extract files from example.deb with filenames matching the /data\..*/ regex
+  #   extract_deb('example.deb')             # extract all files in example.deb
+  #
+  file_found = false if target
+  src_fileIO = File.open(file, 'rb')
+  # get first line of the given file, should be a signature string (`!<arch>\n`) if it is a valid deb file
+  signature = src_fileIO.gets
+
+  abort 'Malformed archive :/' unless signature == "!<arch>\n"
```

Comment:
This seems like an overly simplistic way to determine if an archive is malformed.

Reply:
See below